src/octoprint/util/comm.py

```diff
@@ -2213,10 +2222,31 @@ def _handle_ok(self):
 		if not self._state in self.OPERATIONAL_STATES:
 			return
 
+
+		if (line == "wait"): #reset asynchronos transactions
+			self._AdvancedOkSendNextLines = -1
+		elif ( line is not None ):
+
+			parsed = parse_advanced_ok_line(line)
+			if parsed:
+				self._AdvancedOkSupported = True
+				self._LastProcessedLine = parsed.get("N")
+				free_planner_buff   = parsed.get("P")
+				free_command_buff   = parsed.get("B")
+
+				if (self._LastProcessedLine >= self._SynchronousCommand ):
+					self._AdvancedOkSendNextLines = self._LastProcessedLine + max(free_planner_buff,free_command_buff)
```

Comment:
We should always leave 1 slot free in the command buffer for emergency codes (e.g. M112).

So this should probably rather be 

``` python
min(max(free_planner_buff, free_command_buff - 1), 0)
```

Reply:
i see the problem but your solution will cause overflow, i will think about that

Reply:
How will it cause overflow to send one *less* command than we are allowed to according to the firmware's buffer indicators? 🤔